lib/routes/ehentai/ehapi.js

```diff
@@ -35,7 +51,14 @@ async function parsePage(cache, data, get_bittorrent = false) {
         const el_a = el.find('td[class="gl3c glname"] a');
         const el_img = el.find('td.gl2c div.glthumb div img');
         const title = el_a.find('div.glink').html();
-        const thumbnail = el_img.attr('data-src') ? el_img.attr('data-src') : el_img.attr('src');
+        let thumbnail = el_img.attr('data-src') ? el_img.attr('data-src') : el_img.attr('src');
+        if (config.img_proxy && thumbnail) {
+            const url = new URL(thumbnail);
+            thumbnail = config.img_proxy + url.pathname + url.search;
+        }
+        if (embed_thumb && thumbnail) {
+            thumbnail = await ehgot_thumb(cache, thumbnail);
+        }
         const description = `<img src='${thumbnail}' alt='thumbnail'>`;
         const pubDate = el.find('div[id^="posted_"]').html();
         const link = el_a.attr('href');
```

Comment:
The selector below will match "Front Page Settings" of "Compact" only.
```diff
-                const el_down = el.find('td.gl2c div div.gldown');
+                const el_down = el.find('div.gldown');
```
This may match all 5 "Front Page Settings".